packages/devextreme/testing/testcafe/tests/scheduler/appointments/allDay/allDay.ts

```diff
@@ -86,3 +86,53 @@ test('it should skip weekend days in timelineWorkWeek', async (t) => {
     },
   );
 });
+
+test('should work correctly for unsorted dataSource', async (t) => {
+  const {
+    takeScreenshot,
+    compareResults,
+  } = createScreenshotsComparer(t);
+
+  await t
+    .expect(await takeScreenshot('allDay-unsorted-datasource.png'))
+    .ok();
+
+  await t
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async () => {
+  await createWidget(
+    'dxScheduler',
+    {
+      dataSource: [{
+        id: 2,
+        text: '2',
+        startDate: new Date('2020-11-26T10:30:00.000'),
```

Comment:
Let's change this appointment start date to see it on the screenshot:
E.g. to ```2020-11-26T02:30:00.000```

Reply:
reproduced without this appointment, so I've deleted it